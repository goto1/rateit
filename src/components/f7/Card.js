import React from "react";
import PropTypes from "prop-types";

const Card = ({ className, children }) => {
  const classNames = className ? `card ${className}` : "card";
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

Card.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Card;
