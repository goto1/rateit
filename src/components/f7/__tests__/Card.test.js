import React from "react";
import Card from "../Card";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

const CardContent = (
  <div>
    Lorem ipsum dolor sit amet
    <span>Lorem ipsum dolor sit amet</span>
  </div>
);

describe("Card component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Card>
          {CardContent}
        </Card>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom styling class", () => {
    const wrapper = shallow(
      <Card className="custom-styles">
        {CardContent}
      </Card>
    );
    expect(wrapper).toHaveClassName("custom-styles");
  });
});
