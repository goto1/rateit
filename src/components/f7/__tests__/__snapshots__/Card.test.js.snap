// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<div
  className="card"
>
  <div>
    Lorem ipsum dolor sit amet
    <span>
      Lorem ipsum dolor sit amet
    </span>
  </div>
</div>
`;
